### PR TITLE
HPCC-32305 Soapplus fixes

### PR DESCRIPTION
**Pull request by jakesmith**
Soapplus didn't work correctly with ssl (since at least 9.6)
Use new (correct) semantics for min amount being read from socket
each call.
Simplify code to use a standard ISocket.
Common up persist non-persist read blocks. The non-persist block
did not handle HTTP headers and failed validation.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@mckellyln - please review 1st commit only. It is currently based on the pending socket refactoring PR [https://github.com/hpcc-systems/HPCC-Platform/pull/18841]

File: system/jlib/jsocket.cpp [hunk 1]
```diff
@@ -24,6 +24,23 @@
...
+ * 1. All sockets are created in non-blocking state. All implementations should cope with non-blocking, i.e. wait/retry.
+ * 2. read/readtms will block until min_size bytes are read or timeout expires (see readtms for more detail).
+ * read/readtms can be called with min_size=0, in which case it will return as soon as no more data is available.
+ * 3. Historically, SSL calls + non-SSL write/writetms, used to flip blocking state during the call, this was prone to error,
+ * because socket could [legitimately] be written to whilst being read from, e.g. a server could be writing to a client, and
+ * at the same time, be notified of traffic and start reading it. Flipping the blocking state of a socket is inherently unsafe
+ * in many usage patterns.
+ *
+ * The non-blocking semantics are designed so that the select handlers can read as much data as possible without blocking.
+ * e.g. dafilesrv and MP avoid blocking in their select handlers, thus avoiding up other client traffic.
```
Comment (mckellyln) [inline system/jlib/jsocket.cpp:39]: avoiding **holding** up ...

**Review by mckellyln — COMMENTED** [review]
Looks good. We should (if haven't already) run a bare-metal regression suite with mtls=true for all MP comm to be TLS.

File: system/jlib/jsocket.cpp (hunk 1 above)
Comment (jakesmith) [reply system/jlib/jsocket.cpp]: >@mckellyln - please review 1st commit only. It is currently based on the pending socket refactoring PR [https://github.com/https://github.com/hpcc-systems/HPCC-Platform/pull/18841]

that was part of the jsocket changed not part of this PR. Now merged. I've rebased this PR.
Not worth going issuing a PR to fix the typo' now.

**Comment by jakesmith** [comment]
>Looks good. We should (if haven't already) run a bare-metal regression suite with mtls=true for all MP comm to be TLS.

For the jsocket/securesocket PR (https://github.com/https://github.com/https://github.com/hpcc-systems/HPCC-Platform/pull/18841), I did do that at some point (by pushing a test commit with changes to environment.conf) and let it run in smoketest, but have not done that with latest - but yes should.

**Review by mckellyln — APPROVED** [review]
Looks good to me.

**Pull request merged**